pageserver/src/tenant/mgr.rs

```diff
@@ -184,26 +193,40 @@ impl TenantsMap {
 
     /// A page service client sends a TenantId, and to look up the correct Tenant we must
     /// resolve this to a fully qualified TenantShardId.
+    ///
+    /// During shard splits: we shall see parent shards in InProgress state and skip them, and
+    /// instead match on child shards which should appear in Attached state.  Very early in a shard
+    /// split, or in other cases where a shard is InProgress, we will return our own InProgress result
+    /// to instruct the caller to wait for that to finish before querying again.
     fn resolve_attached_shard(
         &self,
         tenant_id: &TenantId,
         selector: ShardSelector,
-    ) -> Option<TenantShardId> {
+    ) -> ShardResolveResult {
         let mut want_shard = None;
+        let mut any_in_progress = None;
+
         match self {
-            TenantsMap::Initializing => None,
+            TenantsMap::Initializing => ShardResolveResult::NotFound,
```

Comment:
Would be nice to return a fake `InProgress` in the future that will resolve once we transition the TenantsMap away from `Initializing` state

Reply:
Yes... or get rid of Initializing entirely.  That should be possible once we transition all code paths to using a real handle to TenantsManager rather than a static instance.